lib/OpenQA/WebAPI/Plugin/ObsRsync.pm

```diff
@@ -96,15 +110,40 @@ sub _parse_obs_response_dirty {
     my ($res) = @_;
 
     my $results = $res->dom('result');
-    return undef unless $results->size;
+    return (undef, '') unless $results->size;
 
     for my $result ($results->each) {
         my $attributes = $result->attr;
-        return 1 if exists $attributes->{dirty};
+        return (1, 'dirty') if exists $attributes->{dirty};
         next if ($attributes->{repository} // '') ne 'images';
-        return 1 if ($attributes->{state} // '') ne 'published';
+        return (1, $attributes->{state} // '') if ($attributes->{state} // '') ne 'published';
     }
-    return 0;
+    return (0, 'published');
+}
+
+# This method is coupled with openqa-trigger-from-obs and returns
+# string in format %y%m%d_%H%M%S, which corresponds to location
+# used by openqa-trigger-from-obs to determine if any files changed
+# or rsync can be skipped.
+sub _get_run_last_info {
+    my ($c, $project) = @_;
+    my $home = $c->obs_rsync->home;
+    my $run;
+    eval { $run = readlink(Mojo::File->new($home, $project, '.run_last')) };
+    return "" unless $run;
```

Comment:
Do you want to evaluate `$@` for the error message?

Reply:
No, this method is currently used only in cosmetic rendering and I don't know what to do with eventual errors and don't want to flood log on eventual problems.

Reply:
I consider this a dangerous path to tread on. When you can not read the file something could be sevearly broken unless you know what to expect. At the very least I would output `$@`. If, by chance, we later see that there can be expected situations where this makes sense we can still handle the output differently.

Reply:
In this case if something is severely broken many other symptoms will be here for sure, so more spam will not be helpful. At this point I take a risk to go "dangerous path" where some cosmetic fields in some cases are not shown in UI. What you propose is to not implement any features unless we handle all extreme edge cases, while we don't even know if we need that feature at all. It is certainly possible that DB table will be used little later instead of current file-based approach and logic will change drastically, effectively eliminating all work you propose to perform for all edge cases for cosmetic visual effects.